.github/workflows/default.yaml

```diff
@@ -163,9 +163,9 @@ jobs:
 
       - name: Publish build logs
         if: success() || failure()
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
-          name: build-logs-${{ matrix.os }}-${{ matrix.compiler }}
+          name: build-logs-${{ matrix.os }}-${{ matrix.compiler.CC }}-${{ matrix.layer.nick }}
           path: btlayer-*.log
```

Comment:
With this PR changes, each build-tests [artifacts archive](https://github.com/squid-cache/squid/actions/runs/9704296692?pr=1852) contains a single btlayer log file. I think that outcome is acceptable and is better than spending our time on writing configuration/code to merge those individual log files into larger archives, especially since we might eventually add more build artifacts to each layer.